src/agent/onefuzz-supervisor/src/agent.rs

```diff
@@ -100,8 +104,10 @@ impl Agent {
     }
 
     async fn free(&mut self, state: State<Free>) -> Result<Scheduler> {
-        let event = StateUpdateEvent::Free.into();
-        self.coordinator.emit_event(event).await?;
+        if !matches!(self.previous_state, Some(StateValue::Free)) {
```

Comment:
Maybe we should have a method like `emit_state_update_if_changed(new_state: NodeState)`, and just call that in each place we're currently calling `emit`.

Reply:
(and I suggest we just call it everywhere we update the state, not only the states that may currently transition back to themselves)